src/mercury_bulk.c

```diff
@@ -112,12 +114,6 @@ struct hg_bulk {
     hg_uint8_t flags;                    /* Permission flags */
     hg_bool_t eager_mode;                /* Eager transfer */
     hg_atomic_int32_t ref_count;         /* Reference count */
-
-    hg_bool_t bind_addr;                 /* addr binding */
-    void *bulk_hdr;                      /* bulk header */
-    hg_size_t bulk_hdr_size;             /* bulk header size */
```

Comment:
Yes sorry I understood what you were doing, I haven't pushed the related change yet (will let you know when I'm done) but instead of doing that I think we can have some better caching done in mercury_proc_bulk.h and simply point to the serialized copy buffer. What I mean is that if the bulk handle was already serialized and you just received it and you intend to forward it again to a different server, we can just do a memcpy of the entire serialized bulk descriptor (not only the bulk header) since the serialized buffer is still available on the receiver's end.